File: NearbyPlaces/CustomButton.m
//
//  CustomButton.m
//  NearbyPlaces
//
//  Created by Adrian Kubała on 15.02.2017.
//  Copyright © 2017 Adrian Kubała. All rights reserved.
//

#import "CustomButton.h"
#import "CustomView.h"
#import "UIImageView+TintColor.h"

@implementation CustomButton

- (void)drawRect:(CGRect)rect {
  [self drawButton];
  [self setupImageView];
}

- (void)drawButton {
  CustomView *customView = [[CustomView alloc] initWithView:self];
  [customView drawViewByBezierPathWithOffset:5.0 color:[UIColor blackColor]];
}

- (void)setupImageView {
  UIImageView *imgView = [[UIImageView alloc] init];
  imgView.image = [UIImage imageNamed:@"tick"];
  [imgView changeTintColor:[UIColor whiteColor]];
  [self setImage:imgView.image forState:UIControlStateNormal];
}

@end

File: NearbyPlaces/CustomButton.swift
//
//  CustomButton.swift
//  Places
//
//  Created by Adrian on 28.09.2016.
//  Copyright © 2016 Adrian Kubała. All rights reserved.
//

import UIKit

class CustomButton: UIButton {
  override func draw(_ rect: CGRect) {
    drawButton()
    setupImageView()
  }
  
  func drawButton() {
    let customView = CustomView(view: self)
    customView.drawByBezierPath(withOffset: 5.0, color: UIColor.black)
  }
  
  func setupImageView() {
    let imgView = UIImageView()
    imgView.image = UIImage(named: "tick")
    imgView.changeTintColor(UIColor.white)
    setImage(imgView.image, for: UIControlState())
  }
}
